RELEASE.md

```diff
@@ -14,26 +14,39 @@
     * Limitador version the operator enables installations of (without prefix) – i.e. `0.X.Y`
     * If the release is a prerelease
 
-3. Verify that the build [release tag workflow](https://github.com/Kuadrant/dns-operator/actions/workflows/build-images-for-tag-release.yaml) is triggered and completes for the new tag
+3. Verify that the build [release tag workflow](https://github.com/Kuadrant/limitador-operator/actions/workflows/build-images-for-tag-release.yaml) is triggered and completes for the new tag
 
 4. Verify the new version can be installed from the catalog image, see [Verify OLM Deployment](#verify-olm-deployment)
 
-5. Release to the [community operator index catalogs](#community-operator-index-catalogs).
+### Verify OLM Deployment
 
-### Community Operator Index Catalogs
+1. Deploy the OLM catalog image following the [Deploy the operator using OLM](/doc/development.md#deploy-the-operator-using-olm) and providing the generated catalog image. For example:
```

Comment:
This flow could be a bit confusing. It points to a section on how to install olm and the latest of limitador-operator while the command below instructs to install the specific version that the user would be verifying. 
An alternative option could be to have the two commands to create the kind cluster and install olm contained in this one document. wdyt? it's not a lot of duplication. I missed this link first and manually installed olm myself when i realised that the CRDs weren't there. 🙈